go-controller/pkg/ovn/ovn.go

```diff
@@ -682,40 +682,42 @@ func (oc *Controller) WatchNodes() error {
 	return err
 }
 
-// Verify if controller can support ACL logging and validate annotation
-func (oc *Controller) aclLoggingCanEnable(annotation string, nsInfo *namespaceInfo) bool {
-	if !oc.aclLoggingEnabled || annotation == "" {
-		nsInfo.aclLogging.Deny = ""
-		nsInfo.aclLogging.Allow = ""
-		return false
-	}
+// aclLoggingUpdateNsInfo parses the provided annotation values and sets nsInfo.aclLogging.Deny and
+// nsInfo.aclLogging.Allow.
+// *) Valid values for "allow" and "deny" are  "alert", "warning", "notice", "info", "debug".
+// *) Invalid values or the empty string "" will set the logging for the key back to "".
```

Comment:
I was thinking if invalid values should behave as a malformed annotation and return an error. This is one of those things where I  don't have a strong opinion and you can have a different one. Just dropping this here in case you want to consider it.

Reply:
If I return an error there and don't disable logging then we lose idempotency; because we'll keep the log level that we had previously, and all of a sudden the state of logging is not dependent on the current input, but on the input / state that we had before.
So, I really prefer setting this to "" and disabling logging if a user provides invalid input. On the other hand, I will log a warning message indicating the issue to give the admin some feedback.

Reply:
Come to think of it, the same issue obviously is also present with malformed annotation ........

Reply:
I understand the state would not be correctly reflected, but that's more of a consequence of using a single annotation rather than having a separate mechanism to reflect state.
But we wouldn't lose impotency, would we? If the annotation is updated correctly afterwards we would still be able to set it correctly, independently of it's previous (incorrect) value.
Another option we have is to update that annotation back to it's last known correct value. But this also implies some level of confusion for the user, so not sure I would bother with it.

Reply:
> The current approahc where we just keep whatever input we had previously doesn't survive a daemon restart.

With a database wipe in between I guess you mean.

Make sense, and I am ok with this approach.